api/config/v1alpha1/envoygateway_types.go

```diff
@@ -139,11 +139,23 @@ const (
 	RedisBackendType RateLimitDatabaseBackendType = "Redis"
 )
 
-// RateLimitRedisSettings defines the configuration for connecting to
-// a Redis database.
+// RedisTLSSettings defines the TLS configuration for connecting to redis database.
+type RedisTLSSettings struct {
+	// CertificateRef defines the client certificate reference kubernetes secret for TLS connections.
+	//
+	// +kubebuilder:validation:Required
```

Comment:
Have you tried using this config ?

I feel like there are two cases for TLS
1. simple TLS where only TLS needs to be set (no certs need to be specified), and the system CA Cert is used to validate server chain
2. mutual TLS where where certs + CA cert needs to be specified since it's likely that the CA cert for this case is a custom one.

Reply:
user can set tls and env replace some field. Using tls.key tls.crt by default.

Reply:
In fact, ca.crt is needs but the user can also replace it, so i rm it for your suggestion

Reply:
my bad
can we add
```
CACertificateRef *gwapiv1b1.SecretObjectReference `json:"caCertificateRef,omitempty"`
```
as well